python/ribasim/ribasim/input_base.py

```diff
@@ -90,6 +93,53 @@ def _fields(cls) -> list[str]:
         """Return the names of the fields contained in the Model."""
         return list(cls.model_fields.keys())
 
+    def model_dump(self, **kwargs) -> dict[str, Any]:
+        return super().model_dump(serialize_as_any=True, **kwargs)
+
+    # From Pydantic itself, edited to remove the comparison of private attrs
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, BaseModel):
+            self_type = self.__pydantic_generic_metadata__["origin"] or self.__class__
+            other_type = (
+                other.__pydantic_generic_metadata__["origin"] or other.__class__
+            )
+
+            # Perform common checks first
+            if not (
+                self_type == other_type
+                # This comparison has been removed, otherwise we recuse because
+                # we store the parent of the model in a private attribute
+                # and getattr(self, "__pydantic_private__", None)
+                # == getattr(other, "__pydantic_private__", None)
+                and self.__pydantic_extra__ == other.__pydantic_extra__
+            ):
+                return False
+
+            if self.__dict__ == other.__dict__:
+                return True
+
+            model_fields = type(self).__pydantic_fields__.keys()
+            if (
+                self.__dict__.keys() <= model_fields
+                and other.__dict__.keys() <= model_fields
+            ):
+                return False
+
+            getter = (
+                operator.itemgetter(*model_fields)
+                if model_fields
+                else lambda _: pydantic._utils._SENTINEL  # type: ignore
+            )
+            try:
+                return getter(self.__dict__) == getter(other.__dict__)
+            except KeyError:
+                self_fields_proxy = pydantic._utils.SafeGetItemProxy(self.__dict__)  # type: ignore
+                other_fields_proxy = pydantic._utils.SafeGetItemProxy(other.__dict__)  # type: ignore
+                return getter(self_fields_proxy) == getter(other_fields_proxy)
```

Comment:
Could you add some comments here, because I can't follow.
If their `__dict__` is not equal, they can still be equal?

Reply:
You're reviewing Pydantic code, let's not do that. I will clarify this with a better (license) header.